pkg/controller/machineautoscaler/machinetarget_test.go

```diff
@@ -375,3 +377,100 @@ func TestWarningForInvalidGPUAcceleratorLabel(t *testing.T) {
 		})
 	}
 }
+
+// Updating multiple annotations
+func TestUpdatingAnnotations(t *testing.T) {
+	testConfigs := []struct {
+		name                  string
+		expectedNewAnnotation map[string]string
+		suppliedAnnotations   map[string]string
+	}{
+		{
+			name: "Supplying new CPU and Memory annotations",
+			expectedNewAnnotation: map[string]string{
+				annotationsutil.CpuKey:    "1",
+				annotationsutil.MemoryKey: "4Gi",
+			},
+			suppliedAnnotations: map[string]string{
+				annotationsutil.CpuKey:    "1",
+				annotationsutil.MemoryKey: "4Gi",
+			},
+		},
+		{
+			name: "Supplying old CPU annotation and adding new one",
+			expectedNewAnnotation: map[string]string{
+				annotationsutil.CpuKey:           "1",
+				annotationsutil.CpuKeyDeprecated: "1",
+			},
+			suppliedAnnotations: map[string]string{
+				annotationsutil.CpuKeyDeprecated: "1",
+			},
+		},
+		{
+			name: "Supplying old GPU annotation and adding new one",
+			expectedNewAnnotation: map[string]string{
+				annotationsutil.GpuCountKey:           "1",
+				annotationsutil.GpuCountKeyDeprecated: "1",
+				annotationsutil.GpuTypeKey:            "1",
+			},
+			suppliedAnnotations: map[string]string{
+				annotationsutil.GpuCountKeyDeprecated: "1",
+				annotationsutil.GpuTypeKey:            "1",
+			},
+		},
+		{
+			name: "Supplying old max pods annotation and adding new one",
+			expectedNewAnnotation: map[string]string{
+				annotationsutil.MaxPodsKey:           "1",
+				annotationsutil.MaxPodsKeyDeprecated: "1",
+			},
+			suppliedAnnotations: map[string]string{
+				annotationsutil.MaxPodsKeyDeprecated: "1",
+			},
+		},
+	}
+
+	for _, tc := range testConfigs {
+		t.Run(tc.name, func(t *testing.T) {
+			target := NewTarget()
+			target.SetAnnotations(tc.suppliedAnnotations)
+			err := target.UpdateScaleFromZeroAnnotations()
+			if err != nil {
+				t.Errorf("Unexpected error updating ScaleFromZero annotations :%v", err)
+			}
+
+			observed := target.GetAnnotations()
+
+			if !maps.Equal(observed, tc.expectedNewAnnotation) {
+				t.Errorf("SetAnnotations() returned %v, expected %v", observed, tc.expectedNewAnnotation)
+			}
+		})
+	}
+}
+
+// test checkscalefromzeroannotations on a new machine
+func TestScaleFromZeroAnnotations(t *testing.T) {
```

Comment:
@elmiko Do you think it's worth adding this as a test case above to make things neater, or is it fine to leave it out here on it's own?

Reply:
this seems like we are testing the same functionality as above, i would add it as a case there.

Reply:
It actually seems like we are doing what I was trying to test there above on line 389, so I've removed that function entirely :)